Sources/BraintreeThreeDSecure/BTThreeDSecureV2Provider.swift

```diff
@@ -141,18 +129,9 @@ extension BTThreeDSecureV2Provider: CardinalValidationDelegate {
                 errorCode = BTThreeDSecureError.failedAuthentication("").errorCode
             }
             apiClient.sendAnalyticsEvent(BTThreeDSecureAnalytics.challengeFailed)
-            notifyError(
-                withDomain: BTThreeDSecureError.errorDomain,
-                errorCode: errorCode,
-                errorUserInfo: userInfo,
-                completion: completionHandler
-            )
+            completionHandler(nil, NSError(domain: BTThreeDSecureError.errorDomain, code: errorCode, userInfo: userInfo))
         case .cancel:
-            notifyError(
-                withDomain: BTThreeDSecureError.errorDomain,
-                errorCode: BTThreeDSecureError.canceled.errorCode,
-                completion: completionHandler
-            )
+            completionHandler(nil, BTThreeDSecureError.canceled)
```

Comment:
This refactor pointed out a flaw in this logic. As we weren't _actually_ sending the `canceled` error here and instead constructing an error similar to cancel, merchants would need to check for both the code and the raw `canceled` error. Instead above we can construct the `NSError` for the error case, but should have been sending our cancel error here. Without this updated logic we actually don't fire our cancel analytics event in `BTThreeDSecureClient`.

Reply:
Good catch! I actually think that resolves this ticket too (DTBTSDK-3276). I also noticed we don't have a proper unit test for the cancel case from cardinal, which I included in that ticket. Maybe we update that ticket to be unit test specific since your PR here fixes the bug?

Reply:
Ah - nice! Yeah this should resolve that issue. I think we can make the ticket you created test specific and include this fix here. Is it worth calling out as a separate CHANGELOG point?

Reply:
is there any chance we can include the test coverage work in this PR, since we're making the change now? I'm really leery about skipping tests for another day...

Reply:
I poked around at testing this and it actually looks like _nothing_ in this method is tested. Mainly because it involves testing `CardinalValidationDelegate` which uses a class with no inits `CardinalSession`. I can't figure out a good way to test this closed class `CardinalSession` which is required for a mock of this class. @scannillo not sure if you have any other ideas on testing this but imo it seems like it'd be a really large initiative outside of the scope of this PR.